guava/src/com/google/common/hash/AbstractHasher.java

```diff
@@ -50,18 +53,23 @@ public Hasher putUnencodedChars(CharSequence charSequence) {
     return this;
   }
 
+  @SuppressWarnings("argument.type.incompatible")/* Since `charSequence` has min length of 1,
+  `charSequence.toString().getBytes(charset)` return byte array with min length of 1 */
   @Override
-  public Hasher putString(CharSequence charSequence, Charset charset) {
+  public Hasher putString(@MinLen(1) CharSequence charSequence, Charset charset) {
     return putBytes(charSequence.toString().getBytes(charset));
   }
 
+  @SuppressWarnings("upperbound:argument.type.incompatible")/* `off` is required to be @LTLengthOf(value = "#1", offset = "#3 - 1"). Since `bytes.length` min value is 1,
```

Comment:
I'd avoid the `#3` syntax in comments. Say the name of the actual argument instead so that the reader doesn't have to go look it up.